pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,293 @@
+package sysregistriesv2
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/pkg/errors"
+)
+
+// defaultShortNameMode is the default mode of registries.conf files if the
+// corresponding field is left empty.
+const defaultShortNameMode = types.ShortNameModePermissive
+
+// userShortNamesFile is the user-specific registries.conf file to store aliases.
+var userShortNamesFile = filepath.FromSlash(".config/containers/short-name-alias.conf")
+
+// shortNameAliasConf is a subset of the `V2RegistriesConf` format.  It's used in the
+// software-maintained `userShortNamesFile`.
+type shortNameAliasConf struct {
+	// A map for aliasing short names to their fully-qualified image
+	// reference counter parts.
+	Aliases map[string]string `toml:"aliases"`
+
+	namedAliases map[string]reference.Named
```

Comment:
Non-blocking? I’m not much a fan of mixing external formats and internal caches in a single type. (Are we certain the TOML unmarshal won’t allow writing into the cache? Even if we were, it is simpler to not even have to ask.)

`shortNameAliasConf.parseAndValidate` can easily return the cached data as a separate return value.

Reply:
Not entirely sure yet. I leave this open to have another look when other parts of the code are done cooking.